### PR TITLE
Update tutorial script to include immutability and pipelines

### PR DESCRIPTION
**Pull request by cartermp**
I've attempted to add two sections: a short one on Immutability, and a longer one that progressively moves a function towards pipelines and function composition.  I try to leave out gory details as much as possible, as I think it's best to defer to the documentation there (links are provided).

cc @forki @dsyme

### CODE REVIEW

File: vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsx [hunk 1]
```diff
@@ -192,6 +213,75 @@ module Tuples =
...
+    ///
+    /// The result of 'squareOddValuesAndAddOneComposition' is itself another function which takes a
+    /// list of integers as its input.  If you execute 'squareOddValuesAndAddOneComposition' with a list
+    /// of integers, you'll notice that it produces the same results as previous functions.
+    ///
+    /// This is using what is known as function composition.  This is possible because functions in F#
+    /// use Partial Application and the input and output types of each data processing operation match
+    /// the signatures of the functions we're using.
+    let squareOddValuesAndAddOneComposition =
+        List.filter isOdd >> List.map(fun x -> x |> square |> addOne)
```
Comment (saul) [inline vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsx]: May be worth demonstrating:

```f#
List.filter isOdd >> List.map (square >> addOne)
```

**Comment by saul** [comment]
Looks great 👍

File: vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsx (hunk 1 above)
Comment (cartermp) [inline vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsx]: Good feedback!

File: vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsx [hunk 2]
```diff
@@ -64,10 +64,32 @@ module IntegersAndNumbers =
     /// This is a list of all tuples containing all the numbers from 0 to 99 and their squares.
     let sampleTableOfSquares = [ for i in 0 .. 99 -> (i, i*i) ]
 
-    // The next line prints a list that includes tuples, using %A for generic printing.
+    // The next line prints a list that includes tuples, using '%A' for generic printing.
     printfn "The table of squares from 0 to 99 is:\n%A" sampleTableOfSquares
 
 
+/// Values in F# are immutable by default (save for Arrays).  They cannot be changed
```
Comment (dsyme) [inline vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsx]: "save for" sounds slightly archaic and might confuse people who only know a little English.  Perhaps just remove the "(save for Arrays)", it seems a bit out of place to have the exclusion in the intro senetence any case

File: vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsx [hunk 3]
```diff
@@ -192,6 +214,78 @@ module Tuples =
     printfn "Struct Tuple: %A\nReference tuple made from the Struct Tuple: %A" sampleStructTuple (sampleStructTuple |> convertFromStructTuple)
 
 
+/// The F# pipe operators ('|>', '<|', etc.) and F# composition operators ('>>', '<<')
+/// are used extensively when processing data.  These operators are themselves functions
+/// which make use of Partial Application.
```
Comment (dsyme) [inline vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsx]: I think Partial Application need to be capitalized - if anything put it in quotes?  Either is OK as long as style in the document is consistent.

File: vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsx (hunk 2 above)
Comment (cartermp) [inline vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsx]: Good feedback.

File: vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsx (hunk 3 above)
Comment (cartermp) [inline vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsx]: I've capitalized it throughout the document, and also have a link to the section in our docs which explains it.

Comment (dsyme) [inline vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsx]: Typo in my comment - I meant "need not be capitalized".  Sorry.  It looks somehow wrong, as if Making It Mysterious

Comment (cartermp) [inline vsintegration/ProjectTemplates/TutorialProject/Template/Tutorial.fsx]: Okay.  I'll send a quick fix.

**Pull request merged**